<launch>
    <!-- Start odom_to_tf node with parameters -->
    <node name="odom_to_link" pkg="second_project" type="odom_to_base_link" output="screen">
        <param name="child_frame" value="/base_link" />
        <param name="root_frame" value="/odom" />
        <remap from="input_odom" to="/ugv/odom" />
    </node>

    <!-- Lidar header changed to base_link  -->
    <node name="lidar_remap" pkg="second_project" type="lidar_remap" output="screen"/>

    <!-- Start gmapping node -->
    <node pkg="gmapping" type="slam_gmapping" name="gmapping" output="screen">
        <param name="base_frame" value="base_link"/>
        <param name="map_frame" value="map"/>
        <param name="odom_frame" value="odom"/>
        <remap from="scan" to="/scan"/>
    </node>

    <!-- Start pointcloud_to_laserscan node -->
    <node name="pointcloud_to_laserscan" pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" output="screen">
        <remap from="cloud_in" to="/ugv/rslidar_points"/>
        <remap from="scan" to="/scan"/>
    </node>

    <!-- Start RViz with a specified configuration file -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find second_project)/rviz/config1.rviz" />

    <!-- Simulated time -->
    <param name="/use_sim_time" value="true"/>
</launch>
